package.json

```diff
@@ -14,12 +14,10 @@
   ],
   "babel": {
     "presets": [
-      "es2015",
-      "stage-3",
-      "react"
+      "env"
```

Comment:
this includes all the stage4 plugins right?

Reply:
Do we need `react`? We at least need `transform-react-jsx` for when we publish the library (`src/style.js` need to be transpiled)

Reply:
babel-preset-env doesn’t include `stage-x` plugins, it includes all plugins necessary for the `browserslists` config (default is `> 1%, last 2 versions, Firefox ESR`), which is reasonable I believe.
If people want to support older browsers, they should polyfill themselves (using polyfill.io for example).

`preset-react` is not needed as `src/style.js` does not contain any jsx :)

Reply:
> preset-react is not needed as src/style.js does not contain any jsx :)

oh right, my bad.

> babel-preset-env doesn’t include stage-x plugins, it includes all plugins necessary for the browserslists config (default is > 1%, last 2 versions, Firefox ESR), which is reasonable I believe.
If people want to support older browsers, they should polyfill themselves (using polyfill.io for example).

I am not too concerned about polyfills but syntax. Eg. arrow functions are not supported in IE10 or 9. We aim to support what React supports I guess. Can we make sure that preset-env covers the same browsers that are supported right now? ...at list for the client side code :)

Reply:
preset-env includes more than preset-2015 and transpiles to es5. so there shouldn’t be any concerns about that.

from the `preset-env` doc:

> Without any configuration options, babel-preset-env behaves exactly the same as babel-preset-latest (or babel-preset-es2015, babel-preset-es2016, and babel-preset-es2017 together).

Reply:
For example,

```js
export const findStyles = path => {
  if (isStyledJsx(path)) {
    const { node } = path
    return isGlobalEl(node.openingElement) ? [path] : []
  }

  return path.get('children').filter(isStyledJsx)
}
```

gets transpiled to

```js
var findStyles = exports.findStyles = function findStyles(path) {
  if (isStyledJsx(path)) {
    var node = path.node;

    return isGlobalEl(node.openingElement) ? [path] : [];
  }

  return path.get('children').filter(isStyledJsx);
};
```

Reply:
Ah ok, yeah I think that without config includes the stage 4 features. If you have a sec maybe you can transpile master and this branch and diff to see whether there are major differences. If not don't worry I'll take a look asap (tomorrow probably) myself.

Reply:
Here’s a diff: https://gist.github.com/tusbar/2b59d321d9ab4be29582e6c2795fd8ab/revisions

Reply:
Thanks man! Looks good, the only issue I see with that is that Map is not polyfilled now.
Can we include that polyfill only? I would like to avoid to release a major version only because of this.

Reply:
`get-iterator` / `Symbol`s too probably?